package com.mohistmc.silkard.mixin.vanilla.world.entity.projectile;

import net.minecraft.world.entity.projectile.ThrowableProjectile;
import net.minecraft.world.phys.HitResult;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.injection.At;
import org.spongepowered.asm.mixin.injection.Redirect;

@Mixin(ThrowableProjectile.class)
public class MixinThrowableProjectile {

    @Redirect(method = "tick", at = @At(value = "INVOKE", target = "Lnet/minecraft/world/entity/projectile/ThrowableProjectile;hitTargetOrDeflectSelf(Lnet/minecraft/world/phys/HitResult;)V"))
    private void silkard_preHitTargetOrDeflectSelf(ThrowableProjectile projectile, HitResult hitResult) {
        projectile.preHitTargetOrDeflectSelf(hitResult);
    }
}
